src/uniqWith.ts

```diff
@@ -0,0 +1,63 @@
+import { purry } from './purry';
+import { LazyResult, _reduceLazy } from './_reduceLazy';
+import { _toLazyIndexed } from './_toLazyIndexed';
+
+type IsEquals<T> = (a: T, b: T) => boolean;
+
+/**
+ * Returns a new array containing only one copy of each element in the original list.
+ * @param array
+ * @param isEquals the comparator
+ * @signature
+ *    R.uniqWith(array, isEquals)
+ * @example
+ *    R.uniqWith([1, 2, 2, 5, 1, 6, 7], R.equals) // => [1, 2, 5, 6, 7]
```

Comment:
This example is not really useful, because you could just use `R.unique` as well. How about trying to find unique objects that wouldn't be found with `R.unique` because it does a triple equals comparison?

also, the tests should likely also test something like that.

Reply:
Thank you for suggestion. I'll fix that here and in #180

Reply:
Fixed